xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/PVR.h

```diff
@@ -0,0 +1,3076 @@
+  /// name of the associated add-on, e.g. <b>`"inputstream.adaptive.manifest_update_parameter"`</b>.
+  ///
+  ///
+  ///---------------------------------------------------------------------------
+  ///
+  /// **Example:**
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// ...
+  /// PVR_ERROR CMyPVRInstance::GetRecordingStreamProperties(const kodi::addon::PVRRecording& recording,
+  ///                                                        std::vector<kodi::addon::PVRStreamProperty>& properties)
+  /// {
+  ///   ...
+  ///   properties.emplace_back(PVR_STREAM_PROPERTY_INPUTSTREAM, "inputstream.adaptive");
+  ///   properties.emplace_back("inputstream.adaptive.manifest_type", "mpd");
+  ///   properties.emplace_back("inputstream.adaptive.manifest_update_parameter", "full");
+  ///   properties.emplace_back(PVR_STREAM_PROPERTY_MIMETYPE, "application/xml+dash");
+  ///   return PVR_ERROR_NO_ERROR;
+  /// }
+  /// ...
+  /// ~~~~~~~~~~~~~
+  ///
+  virtual PVR_ERROR GetRecordingStreamProperties(
+      const kodi::addon::PVRRecording& recording,
+      std::vector<kodi::addon::PVRStreamProperty>& properties)
+  {
+    return PVR_ERROR_NOT_IMPLEMENTED;
+  }
+  //----------------------------------------------------------------------------
+
+  //==========================================================================
+  /// @ingroup cpp_kodi_addon_pvr_Recordings
+  /// @brief Call one of the recording related menu hooks (if supported).
+  ///
+  /// Supported @ref cpp_kodi_addon_pvr_Defs_Menuhook_PVRMenuhook instances have to be added in
+  /// `constructor()`, by calling @ref AddMenuHook() on the callback.
+  ///
+  /// @copydetails cpp_kodi_addon_pvr_Defs_Menuhook_PVRMenuhook_Help
+  ///
+  ///--------------------------------------------------------------------------
+  ///
+  /// @param[in] menuhook The hook to call.
+  /// @param[in] item The selected recording item for which the hook was called.
+  /// @return @ref PVR_ERROR_NO_ERROR if the hook was called successfully.
+  ///
+  virtual PVR_ERROR CallRecordingMenuHook(const kodi::addon::PVRMenuhook& menuhook,
+                                          const kodi::addon::PVRRecording& item)
+  {
+    return PVR_ERROR_NOT_IMPLEMENTED;
+  }
+  //----------------------------------------------------------------------------
+
+  //============================================================================
+  /// @ingroup cpp_kodi_addon_pvr_Recordings
+  /// @brief **Callback to Kodi Function**\n
+  /// Display a notification in Kodi that a recording started or stopped on the
+  /// server.
+  ///
+  /// @param[in] recordingName The name of the recording to display
+  /// @param[in] fileName The filename of the recording
+  /// @param[in] on True when recording started, false when it stopped
+  ///
+  /// @remarks Only called from addon itself
+  ///
+  inline void Recording(const std::string& recordingName, const std::string& fileName, bool on)
```

Comment:
May I suggest to rename this method to something more meaningful? `Recording()` tells me nothing and the documentation says that this is a callback for a notification from a recording. So, how about naming this method `RecordingNotification`?

Reply:
Makes sense and becomes changed to the RecordingNotification